src/main/java/net/liquidpineapple/pang/objects/DropRandomizer.java

```diff
@@ -11,14 +14,46 @@
 public class DropRandomizer {
 
     private Random randomizer;
+    private final int bronzeNum = 30;
+    private final int silverNum = 20;
+    private final int goldNum = 10;
+    private final int gemNum = 8;
+    private final int heartNum = 2;
+    private final int nullNum = 100-heartNum-gemNum-goldNum-silverNum-bronzeNum;
+
+    private ArrayList<Drop> randomList;
+
 
     public DropRandomizer(){
         Logger.info("Randomizer starting.");
+        randomList = new ArrayList<>();
         randomizer = new Random();
+
+        for(int i=0 ;i<bronzeNum; i++){
+            randomList.add(new Drop("/sprites/drops/coinBronze.png", 0, 0, 0, 1, 10));
+        }
+        for(int i=0 ;i<silverNum; i++){
+            randomList.add(new Drop("/sprites/drops/coinSilver.png", 0, 0, 0, 1, 100));
+        }
+        for(int i=0 ;i<goldNum; i++){
+            randomList.add(new Drop("/sprites/drops/coinGold.png", 0, 0, 0, 1, 1000));
+        }
+        for(int i=0 ;i<(gemNum/4); i++) {
+            randomList.add(new Drop("/sprites/drops/gemBlue.png", 0, 0, 0, 1, 1500));
+            randomList.add(new Drop("/sprites/drops/gemRed.png", 0, 0, 0, 1, 1500));
+            randomList.add(new Drop("/sprites/drops/gemYellow.png", 0, 0, 0, 1, 1500));
+            randomList.add(new Drop("/sprites/drops/gemGreen.png", 0, 0, 0, 1, 1500));
+        }
+        for (int i = 0; i < heartNum; i++) {
+            randomList.add(new Drop("/sprites/drops/heart.png", 0, 0, 0, 1, 1, 1));
+        }
+        for (int i = 0; i < nullNum; i++) {
+            randomList.add(null);
```

Comment:
It is bad practice to add a null value to an array, it would be better to insert a specific value for which you can check.